core/src/main/java/org/testcontainers/lifecycle/TestDescription.java

```diff
@@ -0,0 +1,20 @@
+package org.testcontainers.lifecycle;
+
+import java.util.Optional;
+
+public interface TestDescription {
```

Comment:
I wonder if we need all methods as the public interface. 
The way `BrowserWebDriverContainer` works with it is kind of awkward, implementing the fallback if `getFilesystemFriendlyName()` is not implemented itself?

Reply:
What would you suggest? :)

Reply:
I think it makes sense to not have `getFilesystemFriendlyName` in here, since it seems oddly use case specific?

Or if we want to make it convenient, we could maybe have the default implementation here?
`return getNameParts().map(it -> String.join("-", it));`
And then the default implementation of `getNameParts()` would also become
```
return Optional.of(new String[]{
                    getTestId()
                });
```

Reply:
> oddly use case specific

But we have this use case **at least** in one place.

> default implementation 

I don't think this is a good default implementation to be honest, I would rather keep it explicit and let framework adapters **think** about it, what's the best way to provide a FS friendly name of the test

Reply:
fine by me :)